research/picai/monai_scripts/auto3dseg.py

```diff
@@ -0,0 +1,153 @@
+"""Uses Auto3dSeg from monai to train a segmentation model
+
+-Assumes dataset is formatted following nnUNet structure for consistency
+"""
+
+import argparse
+import os
+from os.path import join
+from pathlib import Path
+from typing import Dict, Optional
+
+import numpy as np
+import yaml
+from batchgenerators.utilities.file_and_folder_operations import save_json
+from monai.apps.auto3dseg.auto_runner import AutoRunner
+
+
+def gen_dataset_list(data_dir: Path, output_path: Optional[str] = None, ext: str = ".nii.gz") -> str:
+    """
+    Generates a MONAI dataset list for an nnUNet structured dataset
+
+    Note: Rather than having a single image and label, this checks for multiple \
+    channels following the nnunet dataset formatting guidelines, and passes a list \
+    of filepaths for each channel as the value for the image key
+
+    Args:
+        data_dir: Path to the nnUNet_raw dataset.
+        output_path: Where and what to save the file as. Must be a json.
+            Default is to save as datalist.json in the data_dir
+
+    Returns:
+        str: The path to where the datalist file was saved
+    """
+
+    train_dir = join(data_dir, "imagesTr")
+    test_dir = join(data_dir, "testTr")
+
+    # Not sure if it is more efficient to figure num channels this way or to just load the dataset.json
+    # and get it from there
+    channels = np.unique([int(file.split(".")[0][-4:]) for file in os.listdir(train_dir)])
```

Comment:
Perhaps the dataset.json approach would be a bit cleaner? If we can just get the key in the file and and check the length it may be a bit more clear whats going on and more efficient as you mentioned. I could be missing something though